packages/opentelemetry-core/src/platform/node/environment.ts

```diff
@@ -23,15 +22,31 @@ import {
 } from '../../utils/environment';
 
 /**
- * Gets the environment variables
+ * Gets the environment variables.
+ * Used directly ONLY in test files.
+ * Accessing env vars is done by getEnv().
  */
-export function getEnv(): Required<ENVIRONMENT> {
+export function calculateEnv(): Required<ENVIRONMENT> {
   const processEnv = parseEnvironment(process.env as RAW_ENVIRONMENT);
-  return Object.assign(
-    {
-      HOSTNAME: os.hostname(),
-    },
-    DEFAULT_ENVIRONMENT,
-    processEnv
-  );
+  const mergedEnv = {
+    ...DEFAULT_ENVIRONMENT,
+    ...processEnv,
+  };
+
+  return mergedEnv;
+}
+
+let calculateEnvCache: Required<ENVIRONMENT>;
```

Comment:
I believe a name that aligns with `parseEnvironment` is better here.

```suggestion
let parsedEnvCache: Required<ENVIRONMENT>;
```